OlinkAnalyze/R/olink_Pathway_Heatmap.R

```diff
@@ -0,0 +1,79 @@
+#' Creates a heatmap of selected pathways and proteins
+#'
+#' Creates a heatmap of proteins related to pathways using enrichment results from olink_pathway_enrichment.
+#'
+#'@param enrich_results data frame of enrichment results from olink_pathway_enrichment()
+#'@param test_results filtered results from statistical test with Assay, OlinkID, and estimate columns
+#'@param method method used in olink_pathway_enrichment ("GSEA" (default) or "ORA")
+#'@param keyword (optional) keyword to filter enrichment results on, if not specified, displays top terms
+#'@param number_of_terms number of terms to display, default is 20
+#'@return A heatmap as a ggplot object
+#'@examples
+#'\donttest{
+#'library(dplyr)
+#' npx_df <- npx_data1 %>% filter(!grepl('control',SampleID, ignore.case = TRUE))
+#' ttest_results <- olink_ttest(df=npx_df,
+#'                              variable = 'Treatment',
+#'                              alternative = 'two.sided')
+#'
+#' try({ # This expression might fail if dependencies are not installed
+#' gsea_results <- olink_pathway_enrichment(data = npx_data1, test_results = ttest_results)
+#' ora_results <- olink_pathway_enrichment(data = npx_data1,
+#'                                         test_results = ttest_results, method = "ORA")
+#' olink_pathway_heatmap(enrich_results = gsea_results, test_results = ttest_results)
+#' olink_pathway_heatmap(enrich_results = ora_results, test_results = ttest_results,
+#'                       method = "ORA", keyword = "cell")
+#' })
+#'
+#'
+#'}
+#'
+#' @seealso \itemize{
+#' \item{\code{\link[OlinkAnalyze:olink_pathway_enrichment]{olink_pathway_enrichment}} for generating enrichment results}
+#' \item{\code{\link[OlinkAnalyze:olink_pathway_visualization]{olink_pathway_visualization}} for generating a bar graph of results}
+#' }
+#'@importFrom dplyr filter distinct inner_join arrange select distinct pull
+#'@importFrom ggplot2 ggplot geom_tile geom_text aes xlab ylab theme
+#'@importFrom stringr str_trunc
+#'@importFrom magrittr %>%
+#'@export
+
+olink_pathway_heatmap<- function(enrich_results, test_results, method = "GSEA", keyword = NULL, number_of_terms = 20){
+  if(is.null(keyword)){
+    sub_enrich <- enrich_results %>% head(number_of_terms)
```

Comment:
I would recommend re-sorting enrich_results just in case.

```suggestion
    sub_enrich <- enrich_results %>% dplyr::arrange(dplyr::desc(estimate)) %>% head(number_of_terms)
```